packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/utils/tile_overlay.dart

```diff
@@ -0,0 +1,23 @@
+import '../types.dart';
+
+/// Converts an [Iterable] of TileOverlay in a Map of TileOverlayId -> TileOverlay.
+Map<TileOverlayId, TileOverlay> keyTileOverlayId(
```

Comment:
I'd avoid adding this to the public API and keep it as a _private method in tile_overlay_updates.dart

Reply:
both methods will need to be used in google_map.dart on the plugin side.
However, we can hide both of them into `TileOverlayUpdates.toJson()` so google_maps.dart only talks to `TileOverlayUpdates` but not these methods directly. WDYT?

I just thought the current style matches how the plugin was built currently. See https://github.com/flutter/plugins/blob/master/packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/utils/polygon.dart